refactor(forms): make FormBuilder classes provided in root

sr5434 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/48237


## What is the new behavior?
formBuilder is provided in root now.
## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

AndrewKushnir commented on aio/content/examples/reactive-forms/src/app/profile-editor/profile-editor.component.2.ts:
```diff
@@ -1,8 +1,6 @@
 // #docplaster
 // #docregion form-builder
 import { Component } from '@angular/core';
-// #docregion form-builder-imports
-import { FormBuilder } from '@angular/forms';
```
It looks like the changes in this file are not needed. Could you please revert them?

sr5434 replied:
OK

AndrewKushnir commented on packages/forms/src/form_builder.ts:
```diff
@@ -107,7 +107,7 @@ export type ɵElement<T, N extends null> =
  * @publicApi
  */
-@Injectable({providedIn: ReactiveFormsModule})
+@Injectable({providedIn: 'root'})
```
There are a couple other symbols in this file that use `providedIn: ReactiveFormsModule`: `UntypedFormBuilder` and `NonNullableFormBuilder`. Could you please update them as well?

AndrewKushnir reviewed (commented):
@sr5434 thanks for addressing the feedback. It looks like the `aio/content/examples/reactive-forms/src/app/profile-editor/profile-editor.component.2.ts` file changes are still there.

Also, please merge all commits into one and make sure the commit message follows the [guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit). In this case, the commit message may look like this:

```
refactor(forms): make FormBuilder classes provided in root

This commit updates the FormBuilder classes to provide them in root
instead of using a deprecated pattern of providing a service in a specific
module using the `providedIn` syntax.

Closes #48237.
```

sr5434 commented:
> @sr5434 thanks for addressing the feedback. It looks like the `aio/content/examples/reactive-forms/src/app/profile-editor/profile-editor.component.2.ts` file changes are still there.
> 
> Also, please merge all commits into one and make sure the commit message follows the [guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit). In this case, the commit message may look like this:
> 
> ```
> refactor(forms): make FormBuilder classes provided in root
> 
> This commit updates the FormBuilder classes to provide them in root
> instead of using a deprecated pattern of providing a service in a specific
> module using the `providedIn` syntax.
> 
> Closes #48237.
> ```

Done

AndrewKushnir commented:
@sr5434 thanks for the reply. I still see 7 commits in this PR and also there are unnecessary changes to the `aio/content/examples/reactive-forms/src/app/profile-editor/profile-editor.component.2.ts` file that can be reverted. Could you please take a look when you get a chance? Thank you.

sr5434 commented:
> @sr5434 thanks for the reply. I still see 7 commits in this PR and also there are unnecessary changes to the `aio/content/examples/reactive-forms/src/app/profile-editor/profile-editor.component.2.ts` file that can be reverted. Could you please take a look when you get a chance? Thank you.

I think I need to reopen this pull request. Would that be ok?

sr5434 committed:
refactor(forms): make FormBuilder classes provided in root

refactor(forms): make FormBuilder classes provided in root

This commit updates the FormBuilder classes to provide them in root
instead of using a deprecated pattern of providing a service in a specific
module using the `providedIn` syntax.

Closes #48237.

sr5434 commented:
I think I fixed the stuff I broke.

AndrewKushnir commented on packages/forms/src/form_builder.ts:
```diff
@@ -107,7 +107,7 @@ export type ɵElement<T, N extends null> =
  *
  * @publicApi
  */
-@Injectable({providedIn: ReactiveFormsModule})
```
I think the `ReactiveFormsModule` symbol import can be removed as well (the `ReactiveFormsModule` is no longer used in this file).

AndrewKushnir reviewed (commented):
@sr5434 thanks for addressing the feedback. I've just left a comment about potentially unused import.

AndrewKushnir reviewed (approved):
@sr5434 thanks for addressing the feedback! The changes look good 👍

I've started some extra tests in Google's codebase([internal only link](http://test/OCL:493096783:BASE:493096826:1670275734016:598602c1)). Once the tests are completed and no issues found, this PR will be added to the merge queue.

Thank you.

AndrewKushnir commented:
@sr5434 FYI if you are interested in contributing one more PR, there is another place that we can cleanup in the same way:

https://github.com/angular/angular/blob/8e52ca271496b0feebf66b2dc7c8f396b73d61a0/packages/forms/src/directives/radio_control_value_accessor.ts#L43-L44

Thank you.

AndrewKushnir commented:
> Isn't that an internal-only service?

@sr5434 yes, it's an internal service and changing the `providedIn` to `root` would still keep it private (i.e. there will be no way to reference it via public APIs).

AndrewKushnir commented:
This PR was merged into the repository by commit 02691a74bb9889bae42f1c1c37b70c946f322dd5.
